stock_picking_invoicing/models/stock_invoice_state_mixin.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019-Today: Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class StockInvoiceStateMixin(models.AbstractModel):
```

Comment:
IMHO, this one should be externalized in a generic module ... this code would never be reused as this...

Reply:
@rousseldenis I am not shocked by the current design. I didn't understand what you mean, could you elaborate?

Reply:
@rvalyi What I wanted to say is that this part of code could be reused in any other model that wants to implement invoice states without necessary pulling this module.

Reply:
Ok. Well may be some improvement to be made later? Or in v12? I'm a bit worried it delays the merge again with speculations about possible usage...

Reply:
Ok let's merge it after fix is done